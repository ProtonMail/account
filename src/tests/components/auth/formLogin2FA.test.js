import render from 'preact-render-to-string';

import FormLogin2FA from '../../../components/auth/formLogin2FA';
import { renderProvided } from '../../testsHelpers/storeTools';

describe('Testing FromLogin2FA', () => {
    test('without U2F', () => {
        expect(render(<FormLogin2FA twoFactorData={{}} />)).toMatchSnapshot();
    });

    test('with U2F', () => {
        expect(renderProvided(<FormLogin2FA twoFactorData={{ U2F: { bar: 'barbar' } }} />)).toMatchSnapshot();
    });
});
