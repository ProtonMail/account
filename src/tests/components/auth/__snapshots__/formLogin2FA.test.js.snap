// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FromLogin2FA with U2F 1`] = `
<form>
  <fieldset>
    <legend>Choose one of the following 2FA methods</legend>
    <div>
      <label htmlFor="password2FA">2FA code</label>
      <div class="twoFactorOption">
        <input type="text" name="password2FA" id="password2FA" autocapitalize="off"
        autocorrect="off" autocomplete="off" minlength="6" maxlength="8" required
        />
      </div>
    </div>
    <div>
      <label htmlFor="u2fKey">Security key</label>
      <div id="u2fKey" class="twoFactorOption">
        <p>Activate your security key...</p>
      </div>
    </div>
  </fieldset>
  <button>Login</button>
</form>
`;

exports[`Testing FromLogin2FA without U2F 1`] = `
<form>
  <fieldset>
    <legend>Choose one of the following 2FA methods</legend>
    <div>
      <label htmlFor="password2FA">2FA code</label>
      <div class="twoFactorOption">
        <input type="text" name="password2FA" id="password2FA" autocapitalize="off"
        autocorrect="off" autocomplete="off" minlength="6" maxlength="8" required
        />
      </div>
    </div>
  </fieldset>
  <button>Login</button>
</form>
`;
